### ISSUE TITLE
`M-<number>` does not work in magit buffer

### ISSUE THREAD

**Issue by gdkrmr** [syl20bnr/spacemacs#10075]
Using `M-<number>` does not change buffer if the cursor is in a magit buffer.

**Comment by duianto** [comment]
If you check the commands that `M-<number>` calls, by pressing:
`C-h k M-<number>` or `SPC h d k M-<number>`

Then you'll see that the bindings from `M-1` to `M-4` call the commands from:
`magit-section-show-level-1-all` to `magit-section-show-level-4-all`, which sets the level (folding depth) of all sections in the current magit buffer. 

`M-5..9` and `M-0` selects the window number, (`0` selects the neotree window, if it's open, otherwise window `10`).

If you check the `C-h k <number>` bindings, then you'll see that a similar thing happens if one wants to move down 4 lines with `4j`. Because `4` calls `magit-section-show-level-4`, which toggles the current sections level.

I also run into this issue every now and then, since it works differently from other modes.

I'm not sure how this should be handled, but it might be best to rebind the magit section level keys, and keep the `M-<number>` (window selection) and `1..4` (prefix digit argument) bindings consistent in most/all modes.

There probably are magit users that want to use the standard bindings `1..4`, `M-1..4`, so maybe there should be a variable to use the standard magit bindings.

Maybe the section level toggles could be bound to `g 1..4` and `g M-1..4`. Or since there doesn't seem to be any major mode bindings under `SPC m` and `,`, in magit buffers, then maybe `, 1..4` and `, M-1..4` could be used?

Currently the simplest workaround probably is to use `SPC <number>` to select windows from magit buffers.

**Comment by gdkrmr** [comment]
`M-<number>` is deep in my muscle memory so I run into this all the time. 
No idea how this should be handled, in my opinion spacemacs key bindings should trump the magit key bindings, I am not sure about the official spacemacs policy.

**Comment by mullr** [comment]
This is definitely still valid; I run into this all the time.

**Comment by mullr** [comment]
As a workaround, I have this in my `user-config`:
```elisp
  (defun my-magit-section-mode-hook ()
    (define-key magit-section-mode-map (kbd "M-1") nil)
    (define-key magit-section-mode-map (kbd "M-2") nil)
    (define-key magit-section-mode-map (kbd "M-3") nil)
    (define-key magit-section-mode-map (kbd "M-3") nil))

  (add-hook 'magit-section-mode-hook 'my-magit-section-mode-hook)
```

**Comment by smile13241324** [comment]
@gdkrmr I have made a commit to make the usual window switching keys work in all magit buffers on develop. Could you double check and close this issue if it works for you?

**Comment by mullr** [comment]
The commit in question is https://github.com/syl20bnr/spacemacs/commit/4b115905886db190bc722342b24492a01a1ecca4

**Comment by mullr** [comment]
I can confirm that this is fixed as of 9bd9151c7.

**Comment by smile13241324** [comment]
@mullr great I will close the PR then.
